### PR TITLE
Accept colorsA/colorsB parameters to the server endpoint

### PR DESCRIPTION
**Pull request by zudov**
With that PR the following requests would work:
- /badge/foo-bar-baz?colorsA=abcdef&colorsB=black
- /badge/foo-bar-baz?colorsA=abcdef
- /badge/foo-bar-baz?colorsB=black

### CODE REVIEW

File: server.js [hunk 1]
```diff
@@ -4288,13 +4288,14 @@ function(data, match, end, ask) {
...
   try {
     var badgeData = getBadgeData(subject, data);
-    badgeData.colorscheme = undefined;
     if (data.label !== undefined) { badgeData.text[0] = '' + data.label; }
     badgeData.text[1] = status;
-    if (sixHex(color)) {
-      badgeData.colorB = '#' + color;
-    } else {
-      badgeData.colorscheme = color;
+    if ((data.colorB === undefined) && (data.colorA === undefined)) {
```
Comment (espadrine) [inline server.js]: If we had, say, `/badge/so-colourful-fa4391.svg?colorA=009900`, we'd expect to have a somewhat pink right-hand-side, and a somewhat dark green left-hand-side. Instead, this block is not executed, and the right-hand-side is green.

We only want to modify `colorB` (in the `if (sixHex(color)) {…}` branch) if the query parameter `colorB` is not defined, and that's it. In the `else` branch, we only want to set the `colorscheme` if neither `colorA` and `colorB` are defined as a query parameter.

File: server.js [hunk 2]
```diff
@@ -4372,13 +4376,14 @@ function getBadgeData(defaultLabel, data) {
...
   }
 
-  return {
-    text: [label, 'n/a'],
-    colorscheme: 'lightgrey',
-    template: template,
-    logo: data.logo,
-    logoWidth: +data.logoWidth,
-    links: data.link,
+  return new function BadgeData() {
```
Comment (espadrine) [inline server.js]: I am not a fan of this construct. Could you keep using the object literal?

```js
var data = {…};
if (…) { data.colorA = … }
…
return data;
```

**Comment by espadrine** [comment]
Thanks for contributing!

Was this linked to an issue?

On line 185, the cache needs information about the new query parameters.

**Comment by espadrine** [comment]
Are you interested in pursuing this patch, or should someone take over?

**Comment by dhowe** [comment]
status?

**Comment by zudov** [comment]
I rebased to current master, and fixed the mentioned problems. Sorry for such a delay. Could you give it another review?

File: try.html [hunk 3]
```diff
@@ -1032,6 +1032,10 @@ <h2 id="styles"> Styles </h2>
   <tr><td><code>?link=http://abc.xyz&amp;link=http://shields.io</code></td>
     <td>Specify what clicking on the left/right of a badge should do (esp. for
       social badge style)</td></tr>
+  <tr><td><code>?colorA=abcdef<code></td>
```
Comment (espadrine) [inline try.html]: Use `</code>` instead of the second `<code>`.

**Comment by espadrine** [comment]
Rebased, modified and merged by 52b0369c1494cb692fe1f82f74766af46180b47b, modified by 29a4803e4b3d5fe4ecd7af947843a0444c198639.

Thanks for your contribution! This is now live.

**Comment by vszakats** [comment]
Thanks for this. One closing `<code>` remained in `index.html`.

**Comment by espadrine** [comment]
Duly noted, I'll update it when I `make website` next, which should be in about 29h.

**Pull request closed without merging**